tests/src/abi-tests.ts

```diff
@@ -225,3 +232,37 @@ test('Validation fails when abi is missing specified function', () => {
 test('ABI function to repr string', () => {
   expect(abiFunctionToString(TEST_ABI.functions[1])).toEqual('(define-public (hello (arg1 int)))');
 });
+
+test('Parse string input using ABI arg type', () => {
+  const uintString = '123';
+  const intString = '234';
+  const boolStringTrue = 'True';
+  const boolStringFalse = 'False';
+  const standardPrincipalString = 'ST3KC0MTNW34S1ZXD36JYKFD3JJMWA01M55DSJ4JE';
+  const address = 'ST3KC0MTNW34S1ZXD36JYKFD3JJMWA01M55DSJ4JE';
+  const contractName = 'kv-store';
+  const contractPrincipalString = `${address}.${contractName}`;
+  const bufferString = 'test 321';
+
+  const uintFunctionArgType: ClarityAbiType = 'uint128';
+  const intFunctionArgType: ClarityAbiType = 'int128';
+  const boolFunctionArgType: ClarityAbiType = 'bool';
+  const principalFunctionArgType: ClarityAbiType = 'principal';
+  const bufferFunctionArgType: ClarityAbiType = { buffer: { length: 1 } };
```

Comment:
Looks like this should be causing the parsing below to throw since `'test 321'.length > 1`. I'm assuming this is currently just truncating the value to the first byte (`t`)?

Reply:
the buffer length is now checked by the parser